### PR TITLE
add linux 6.1 support

### PR DESCRIPTION
**Pull request by KanjiMonster**
Add linux 6.1 and clean up the configs.

Going from 5.15 Linux 6.1 brings various improvements:

 * various improvements to BPF
 * improvements to MPTCP
 * support for AMT (RFC7450)
 * improved unix socket performance
 * improved UDPv6 performance
 * VxLAN support for terminating configured VNIs only
 * support for TCPv6 segments > 64k (BIG TCP)
 * Ethernet Power Sourcing Equipment and Power Device support
 * bridge: 802.1x support
 * bridge: Multiple Spanning Tree (MST) mode

Kernel config changes are mostly trivial removed or renamed symbols.

One major change is that due to increased depedencies the i915 gpu driver would require enabling more subsystems. None of the switches's SoCs do implement the i915 gpu though, so I opted to drop the driver instead of adding its dependencies.

Updates to external modules were done via coccinelle semantic patches to automate updating them. The semantic patches used are placed into the tree so further modules can be updated as needed.

Since the yocto kernel cache isn't branched yet, we point to master for kmeta. This is done in an extra commit so that moving to the branch can be easily done by reverting that commit.

Depends on https://github.com/bisdn/meta-ofdpa/pull/53.

Signed-off-by: Jonas Gorski <jonas.gorski@bisdn.de>

### CODE REVIEW

File: scripts/coccinelle/modules/6.1-i2c_remove_sig.cocci [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: GPL-2.0-only
```
Comment (ideaship) [inline scripts/coccinelle/modules/6.1-i2c_remove_sig.cocci:1]: Can you add a README.md to `scripts/coccinelle/modules/` that gives some context for this files and explains when and how to use them?

It is not entirely clear from the commit message if all or just some of the patches in "onl: update modules for linux 6.1" have been created by the two semantic patches.

File: recipes-kernel/linux/linux-yocto-onl-linux-6.1.y/arch/arm/boot/dts/arm-accton-as4610-30p.dts [hunk 2]
```diff
@@ -0,0 +1,225 @@
+/*
```
Comment (ideaship) [inline recipes-kernel/linux/linux-yocto-onl-linux-6.1.y/arch/arm/boot/dts/arm-accton-as4610-30p.dts:1]: Commit message "linux-yocto-onl: add linux 6.1 support":
- s/terminating configures VNIs/terminating configured VNIs/
- can you spell out BSP?

Comment (KanjiMonster) [reply recipes-kernel/linux/linux-yocto-onl-linux-6.1.y/arch/arm/boot/dts/arm-accton-as4610-30p.dts:1]: > can you spell out BSP?

I can't from the top of my head (I could guess), but I can extend it to KCONF_BSP_AUDIT, is that enough?

Comment (ideaship) [reply recipes-kernel/linux/linux-yocto-onl-linux-6.1.y/arch/arm/boot/dts/arm-accton-as4610-30p.dts:1]: I was hoping for something that has less than 198.000.000 Google hits, but more than zero.

Comment (ideaship) [reply recipes-kernel/linux/linux-yocto-onl-linux-6.1.y/arch/arm/boot/dts/arm-accton-as4610-30p.dts:1]: I do. I would just like the commit message to be useful to people who don't.

File: scripts/coccinelle/modules/6.1-i2c_remove_sig.cocci (hunk 1 above)
Comment (KanjiMonster) [reply scripts/coccinelle/modules/6.1-i2c_remove_sig.cocci:1]: Added a (smallish) README.md

**Review by ideaship — COMMENTED** [review]
It seems to me that commit e024e2ad that was recently merged into master was accidentally omitted from recipes-kernel/linux/linux-yocto-onl_6.1.bb.

File: recipes-kernel/linux/linux-yocto-onl-linux-6.1.y/bisdn-kmeta/cfg/0001-tun-allow-setting-hwaddr-on-tap-interface-creation.patch [hunk 3]
```diff
@@ -0,0 +1,43 @@
+Upstream-Status: undecided if suitable [rather hackish, but also simple]
```
Comment (ideaship) [inline recipes-kernel/linux/linux-yocto-onl-linux-6.1.y/bisdn-kmeta/cfg/0001-tun-allow-setting-hwaddr-on-tap-interface-creation.patch:1]: This file is missing the change "linux-yocto-onl: move Upstream-Status into commit message" from the 5.15 version.

File: recipes-kernel/linux/linux-yocto-onl-linux-6.1.y/bisdn-kmeta/bsp/armel-iproc/10-11-bgmac-xgs-iproc-changes.patch [hunk 4]
```diff
@@ -0,0 +1,485 @@
...
+ 
+ 	bgmac_enable(bgmac);
+@@ -1195,10 +1202,25 @@ static int bgmac_open(struct net_device *net_dev)
+ 	}
+ 	napi_enable(&bgmac->napi);
+ 
++	if (IS_ENABLED(CONFIG_ARCH_XGS_IPROC)) {
++		/* Reset emulated MIB statistics to zero */
++		memset(&bgmac->estats, 0, sizeof(struct bgmac_ethtool_stats));
++	}
```
Comment (ideaship) [inline recipes-kernel/linux/linux-yocto-onl-linux-6.1.y/bisdn-kmeta/bsp/armel-iproc/10-11-bgmac-xgs-iproc-changes.patch:220]: I see no mention in the commit message of this change compared to the Linux 5.15 version.

File: recipes-kernel/linux/linux-yocto-onl-linux-6.1.y/bisdn-kmeta/bsp/armel-iproc/10-16-usb-ipproc-xgs-hack.patch [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
++#if CONFIG_USB_EHCI_XGS_IPROC
++	hcd->usb_phy = phy;
++#endif
+ 	err = usb_add_hcd(hcd, irq, IRQF_SHARED);
+ 	if (err)
+ 		goto err_power;
+ 
+ 	device_wakeup_enable(hcd->self.controller);
+ 	device_enable_async_suspend(hcd->self.controller);
++
```
Comment (ideaship) [inline recipes-kernel/linux/linux-yocto-onl-linux-6.1.y/bisdn-kmeta/bsp/armel-iproc/10-16-usb-ipproc-xgs-hack.patch:87]: Adding this empty line seems unrelated to updating the patch for 6.1.

File: recipes-kernel/linux/linux-yocto-onl-linux-6.1.y/bisdn-kmeta/bsp/armel-iproc/10-11-bgmac-xgs-iproc-changes.patch (hunk 4 above)
Comment (KanjiMonster) [reply recipes-kernel/linux/linux-yocto-onl-linux-6.1.y/bisdn-kmeta/bsp/armel-iproc/10-11-bgmac-xgs-iproc-changes.patch:220]: As indicated by the single pluses, it's a change in the upstream driver, not in the modifications of the patch.

Comment (KanjiMonster) [reply recipes-kernel/linux/linux-yocto-onl-linux-6.1.y/bisdn-kmeta/bsp/armel-iproc/10-11-bgmac-xgs-iproc-changes.patch:220]: That change was apparently accidentally dropped in https://github.com/bisdn/meta-open-network-linux/commit/351e020ee9c77df4c4b2abbee80a3343ee1fbf59.

File: recipes-kernel/linux/linux-yocto-onl-linux-6.1.y/bisdn-kmeta/cfg/0001-tun-allow-setting-hwaddr-on-tap-interface-creation.patch (hunk 3 above)
Comment (KanjiMonster) [reply recipes-kernel/linux/linux-yocto-onl-linux-6.1.y/bisdn-kmeta/cfg/0001-tun-allow-setting-hwaddr-on-tap-interface-creation.patch]: Fixed.

File: recipes-kernel/linux/linux-yocto-onl-linux-6.1.y/bisdn-kmeta/bsp/armel-iproc/10-16-usb-ipproc-xgs-hack.patch (hunk 5 above)
Comment (KanjiMonster) [reply recipes-kernel/linux/linux-yocto-onl-linux-6.1.y/bisdn-kmeta/bsp/armel-iproc/10-16-usb-ipproc-xgs-hack.patch]: Artifact from rebasing, fixed.

File: recipes-kernel/linux/linux-yocto-onl-linux-6.1.y/bisdn-kmeta/bsp/armel-iproc/10-11-bgmac-xgs-iproc-changes.patch (hunk 4 above)
Comment (ideaship) [reply recipes-kernel/linux/linux-yocto-onl-linux-6.1.y/bisdn-kmeta/bsp/armel-iproc/10-11-bgmac-xgs-iproc-changes.patch:220]: So shouldn't adding it back be a separate commit which fixes the accident for 5.15 as well?

Comment (KanjiMonster) [reply recipes-kernel/linux/linux-yocto-onl-linux-6.1.y/bisdn-kmeta/bsp/armel-iproc/10-11-bgmac-xgs-iproc-changes.patch:220]: I didn't intentionally add it back, I will need to figure out where it got lost for 5.15.

Comment (KanjiMonster) [reply recipes-kernel/linux/linux-yocto-onl-linux-6.1.y/bisdn-kmeta/bsp/armel-iproc/10-11-bgmac-xgs-iproc-changes.patch:220]: Might have been a quick (build) fix I forgot to replace with the actual commit.
